### DISCUSSION TITLE
Link with url containing space won't render in markdown

### DISCUSSION THREAD

**Discussion by HalfSweet** [vuepress-theme-hope/vuepress-theme-hope#3707] [Q&A] [answered]
### Checklist

- [X] I am using **LTS** version of Node.js.
- [X] I am using the **latest v2 version** of VuePress and VuePress official plugins.
- [X] I am using the **latest v2 version** of vuepress-theme-hope and all plugins in this repo.
- [X] I followed the docs and I double checked my configuration.
- [X] I have searched [FAQ](https://theme-hope.vuejs.press/faq/) and I did not find the answer.

### Package name

vuepress-theme-hope

### Which operating system are you using?

- [ ] macOS
- [X] Windows
- [ ] Linux

### Describe the bug

```markdown
[正常](https://example.ex/test0.html)

[错误](https://example.ex/test 1.html)
```
![图片](https://github.com/vuepress-theme-hope/vuepress-theme-hope/assets/60973476/857ae3bc-9c0f-43f9-9b9a-fd41f283dae6)

**Comment by HalfSweet** [comment]
在一些markdown渲染器中，例如 Typora 或者 Writeside 中，可以正常渲染`[链接文本](URL)`中url带空格的情况，不过经过我的查找发现似乎没有做对这种情况的说明…另外我找到了一些规避的方式，使用`[链接文本](<URL>)`即可，但是当我尝试使用`<URL>`的方式来进行书写的时候，发现同样会渲染不完全。但是使用`[链接文本](<URL>)`似乎是完全可行的，我可以在url中包含任何的特殊字符。

感谢您的回复

**Comment by Mister-Hope** [comment, marked as answer]
The solution is to use actual URL, `https://example.com/a b.html` is never a valid one, but `https://example.com/a%20b.html` is.

Marakdown-it playground: https://markdown-it.github.io/#md3=%7B%22source%22%3A%22%5Burl%5D%28https%3A%2F%2Fexample.com%2Fa%20b%29%22%2C%22defaults%22%3A%7B%22html%22%3Afalse%2C%22xhtmlOut%22%3Afalse%2C%22breaks%22%3Afalse%2C%22langPrefix%22%3A%22language-%22%2C%22linkify%22%3Atrue%2C%22typographer%22%3Atrue%2C%22_highlight%22%3Atrue%2C%22_strict%22%3Afalse%2C%22_view%22%3A%22html%22%7D%7D


Commonmark Spec: https://spec.commonmark.org/
